### PR TITLE
Write an integration test for mobilecoind-json

### PR DESCRIPTION
**Pull request by eranrund**
We'd like to test `mobilecoind-json` as part of our Jenkins CD process, this PR provides a Python script that exercises some of `mobilecoind-json`'s core functionality.

### Future Work
* Get this to run as part of a k8 deployment in `internal`

### CODE REVIEW

File: mobilecoind-json/tests/integration_test.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+# Integration test that uses mobilecoind-json to submit a transaction and check balance.
+
+import argparse
+import glob
+import json
+import logging
+import os
+import sys
+import time
+import urllib.request
```
Comment (eranrund) [inline mobilecoind-json/tests/integration_test.py:14]: I know `requests` exists and is more pleasant to use but I wanted something that has no external dependencies.

**Review by iamalwaysuncomfortable — APPROVED** [review]
Looks fine, we definitely need a set of python integration tests for our python code that runs in CI so I'm glad to see this!

File: mobilecoind-json/tests/integration_test.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+            "num_subaddresses": num_subaddresses,
+        })
+
+    def get_monitor(self, monitor_id):
+        return self.request(f"monitors/{monitor_id}")
+
+    def get_ledger_local(self):
+        return self.request(f"ledger/local")
+
+    def wait_for_monitor_to_sync(self, monitor_id, poll_interval=0.1):
```
Comment (remoun) [inline mobilecoind-json/tests/integration_test.py:60]: Nit: This is rather different from the other methods that forward to `self.request()`, so how about we move it below those methods?

File: mobilecoind-json/tests/integration_test.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    return [
+        json.load(open(f)) for f in
+        sorted(glob.glob(os.path.join(keys_dir, "*.json")), key=filename_key)
+    ]
+
+
+def run_test(mobilecoind_json_url, keys_dir, max_seconds):
+    if not mobilecoind_json_url.endswith("/"):
+        mobilecoind_json_url += "/"
+
```
Comment (remoun) [inline mobilecoind-json/tests/integration_test.py:107]: Nit: `MobilecoindJsonClient` does the same check, so let's remove this
```suggestion
```

File: mobilecoind/strategies/drain-accounts.py [hunk 4]
```diff
@@ -54,7 +54,7 @@ def parse_args() -> argparse.ArgumentParser:
     parser.add_argument("--max-seconds",
                         type=int,
                         default=40,
-                        help="Number of seconds to wait for a tx to clearn")
+                        help="Number of seconds to wait for a tx to clean")
```
Comment (remoun) [inline mobilecoind/strategies/drain-accounts.py:57]:
```suggestion
                        help="Number of seconds to wait for a tx to clear")
```

File: mobilecoind/strategies/test_client.py [hunk 5]
```diff
@@ -39,7 +39,7 @@ def parse_args() -> argparse.ArgumentParser:
     parser.add_argument("--max-seconds",
                         type=int,
                         default=40,
-                        help="Number of seconds to wait for a tx to clearn")
+                        help="Number of seconds to wait for a tx to clean")
```
Comment (remoun) [inline mobilecoind/strategies/test_client.py:42]:
```suggestion
                        help="Number of seconds to wait for a tx to clear")
```

**Review by remoun — APPROVED** [review]

**Pull request merged**